samples/javascript_typescript/60.multilingual-luis-bot/translator.ts

```diff
@@ -0,0 +1,345 @@
+			const response = await request({
+				url: uri,
+				method: 'POST',
+				headers: {
+                    'Ocp-Apim-Subscription-Key': this.apiKey,
+					'Content-Type': 'application/json',
+				},
+				body: JSON.stringify(translationRequests),
+            });
+            let translations = JSON.parse(response);
+            translations.forEach((traslationModel, index, array) => {
+                                let translation = traslationModel.translations[0].text;
+                                if(traslationModel.translations[0].alignment){
+                                    let alignment = traslationModel.translations[0].alignment.proj;
+                                    translation = this.postProcessor.fixTranslation(orgTexts[index], alignment, translation, from);
+                                }
+                                let result: TranslationResult = { translatedText: translation };
+                                 results.push(result);
+                            });
+			return results;
+		} catch (e) {
+			switch (e.statusCode) {
+				case 401:
+					throw new Error('Cognitive Authentication Failed');
+				default:
+					throw new Error('Internal Error');
+			}
+		}
+    };
+}
+
+
+/**
+ * @private
+ */
+class TranslatorPostProcessor {
+    noTranslatePatterns: { [languageId : string] : string[]};
+    wordDictionary: { [id: string]: string };
+
+    constructor(noTranslatePatterns?: { [languageId : string] : string[]}, wordDictionary?: { [id: string]: string }) {
+        this.noTranslatePatterns = {};
+        this.wordDictionary = wordDictionary;
+        
+        if (wordDictionary) {
+            Object.keys(this.wordDictionary).forEach(word => {
+                if (word != word.toLowerCase()) {
+                    Object.defineProperty(this.wordDictionary, word.toLowerCase(),
+                        Object.getOwnPropertyDescriptor(this.wordDictionary, word));
+                    delete this.wordDictionary[word];
+                }
+            });
+            
+        }
+
+        if(noTranslatePatterns) {
+            for(var key in noTranslatePatterns) {
+                noTranslatePatterns[key].forEach((pattern, index, array) => {
+                    if(!(key in this.noTranslatePatterns)){
+                        this.noTranslatePatterns[key] = [];
+                    }
+                    if (pattern.indexOf('(') == -1) {
+                        pattern = `(${pattern})`;
+                    }
+                    this.noTranslatePatterns[key][index] = pattern;
+                });
+               
+            }
+        }  
+    }
+
+    private join(delimiter: string, words: string[]): string {
+        return words.join(delimiter).replace(/[ ]?'[ ]?/g, "'").trim();
+    }
+
+    private splitSentence(sentence: string, alignments: string[], isSrcSentence = true): string[] {
+        let wrds = sentence.split(' ');
```

Comment:
```suggestion
        const words = sentence.split(' ');
```
1) Use `const`
2) Use clear variable names